lib/script/script.js

```diff
@@ -1480,38 +1481,53 @@ class Script {
    * @param {Number} m
    * @param {Number} n
    * @param {Buffer[]} keys
+   * @param {Boolean?} isSorted
+   * @param {Boolean?} isWitness
+   * @returns {Script}
    */
 
-  fromMultisig(m, n, keys) {
+  fromMultisig(m, n, keys, isSorted = true, isWitness = true) {
     assert((m & 0xff) === m && (n & 0xff) === n);
     assert(Array.isArray(keys));
     assert(keys.length === n, '`n` keys are required for multisig.');
     assert(m >= 1 && m <= n);
-    assert(n >= 1 && n <= 15);
+
+    assert(
+      n >= 1 && n <= MAX_MULTISIG_PUBKEYS,
+      `${n} keys not allowed in script. Max allowed: ${MAX_MULTISIG_PUBKEYS}`
+    );
 
     this.clear();
 
-    this.pushSmall(m);
+    this.pushInt(m);
 
-    for (const key of sortKeys(keys))
+    for (const key of isSorted ? sortKeys(keys) : keys)
       this.pushData(key);
 
-    this.pushSmall(n);
+    this.pushInt(n);
     this.pushOp(opcodes.OP_CHECKMULTISIG);
 
-    return this.compile();
+    const script = this.compile();
+
+    if (!isWitness) {
+      assert(script.getSize() <= MAX_SCRIPT_PUSH, 'Script size is too large');
```

Comment:
I think the check should be this because it will eventually get wrapped in sh
```suggestion
      assert(script.getSize() + 3 <= MAX_SCRIPT_PUSH,, 'Script size is too large');
```

Reply:
nvm, my bad